--- point_script.d.ts.orig	2026-03-05 01:57:09.604559200 +0100
+++ point_script.d.ts.orig	2026-03-31 09:12:48.466888100 +0200
@@ -214,8 +214,24 @@
         EntFireAtName(name: string, input: string, inputData?: InputValue | { value?: InputValue, caller?: Entity, activator?: Entity }, delay?: number): void;
         /** @deprecated This overload will be removed in a future update */
         EntFireAtTarget(target: Entity, input: string, inputData?: InputValue | { value?: InputValue, caller?: Entity, activator?: Entity }, delay?: number): void;
+
+        // Extensions
+
+        /** Show a hud hint to every player */
+        ShowHudHintAll(text: string, isAlert: boolean): void;
+        /** Listens for a protobuf message originating from the server with the specified (partial) name,
+        * callback can return false to block the message or true to let it through */
+        OnUserMessage(config: { messageName: string, callback: (info: UserMessageInfo) => boolean }): void;
+        CreateUserMessage(messageName: string): UserMessageInfo;
+        /** Emits a specified soundevent by name. If source entity is not provided, it will be played globally. if recipients are not specified, it will be played for all players */
+        EmitSound(config: { soundName: string, source?: Entity, volume?: number, pitch?: number, recipients?: CSPlayerController[] }): void;
+        /** Retrieves value of a convar */
+        GetConVarValue(cvar: string): CvarValue
+        /** Prints a message to the chat, supports special characters for color codes */
+        PrintToChatAll(message: string): void;
     }
 
+    type CvarValue = string | number | boolean | Vector | QAngle | Color | undefined;
     type Vector = { x: number, y: number, z: number };
     type RotationVector = { x: number, y: number, z: number };
     type QAngle = { pitch: number, yaw: number, roll: number };
@@ -506,6 +522,18 @@
         Teleport(newPosition: Vector | null, newAngles: QAngle | null, newVelocity: Vector | null): void;
         /** @deprecated This method will be removed in a future update */
         GetLocalVelcoity(): Vector;
+
+        // Extensions
+
+        SetMoveType(moveType: number): void;
+        /** @experimental Access value of a schema field for this entity
+         * This is currently limited to only atomic fields, meaning that it does not support nested components */
+        GetSchemaField(className: string, fieldName: string): Entity | number | string | boolean | undefined;
+        /** Override the transmission of this entity to the target player 
+         * @note Use responsibly, do not deny transmission of player pawn to it's owner, or weird things might happen, including players crashing. */
+        SetTransmitState(controller: CSPlayerController, state: boolean): void;
+        /** Set whether the entity will be transmitted to all players, will apply to newly connected clients too */
+        SetTransmitStateAll(state: boolean): void;
     }
 
     interface EntityDamage {
@@ -574,6 +602,16 @@
         IsBot(): boolean;
         IsConnected(): boolean;
         JoinTeam(team: number): void;
+
+        // Extensions
+
+        ShowHudHint(text: string, isAlert: boolean): void;
+        /** Shows a hud message supporting HTML syntax for text effects */
+        ShowHudMessageHTML(htmlText: string, duration: number): void;
+        /** Retrieve SteamID64 of this player */
+        GetSteamID(): number;
+        /** Respawns this player's pawn */
+        Respawn(): void;
     }
 
     export class CSObserverPawn extends BaseModelEntity {
@@ -634,6 +672,22 @@
 
     /** @deprecated This enum will be removed in a future update */
     export enum CSDamageType { }
+
+    // Extensions
+
+    export class UserMessageInfo {
+        /** @note accessing repeated fields is not supported yet */
+        GetField<T>(fieldName: string): T;
+        /** @note accessing repeated fields is not supported yet */
+        SetField(fieldName: string, value: any): void;
+
+        RemoveRecipient(playerSlot: number): void;
+        AddRecipient(playerSlot: number): void;
+        AddAllRecipients(): void;
+        ClearRecipients(): void;
+
+        Send(): void;
+    }
 }
 
 /**
